tasks/page/processIBOM.js

```diff
@@ -0,0 +1,42 @@
+const fs = require('fs')
+const globule = require('globule')
+const path = require('path')
+const utils = require('../utils/utils')
+const cp = require('child_process')
+
+if (require.main !== module) {
+  module.exports = function(config, boardInfo) {
+    let kicadPcbFile
+    if (
+      boardInfo.eda &&
+      boardInfo.eda.type === 'kicad' &&
+      boardInfo.eda.pcb != null
+    ) {
+      kicadPcbFile = path.join(boardInfo.repoPath, boardInfo.eda.pcb)
+    } else if (boardInfo.eda == null) {
+      const kicadPcbPattern = path.join(boardInfo.boardPath, '**/*.kicad_pcb')
+      kicadPcbFile = globule.find(kicadPcbPattern)[0]
+    }
+    if (kicadPcbFile == null) {
+      return {deps: [], targets: [], moduleDep: false};
+    }
+    const deps = [kicadPcbFile]
+    const buildFolder = boardInfo.boardPath.replace('boards', 'build/boards')
+    const targets = [
+      `${buildFolder}/interactive_bom.html`
+    ]
+    return {deps, targets, moduleDep: false}
+  }
+} else {
+  const {config, deps, targets} = utils.processArgs(process.argv)
+  let kicadPcbFile = deps[0];
+  let ibom = targets[0];
+  const run_ibom = path.join(__dirname, 'run_ibom')
+  cp.execSync(`${run_ibom} ${kicadPcbFile} ${ibom}`)
+}
+
+function __guard__(value, transform) {
+  return typeof value !== 'undefined' && value !== null
+    ? transform(value)
+    : undefined
+}
```

Comment:
These `__guard__` functions that we use in a few places, were from when I converted the codebase from coffeescript to javascript, we really shouldn't be using them in new code.

Reply:
These are just a left-over from cutting and pasting from `order_pcbs.jsx`. I need to revisit all this. I was just happy enough to get the thing displaying a single button in the right place that looked OK!